Library/Homebrew/test/test_cmd_audit.rb

```diff
@@ -380,4 +380,99 @@ class Foo < Formula
   ensure
     ENV["HOMEBREW_NO_GITHUB_API"] = original_value
   end
+
+  def test_audit_caveats
+    fa = formula_auditor "foo", <<-EOS.undent
+      class Foo < Formula
+        homepage "http://example.com/foo"
+        url "http://example.com/foo-1.0.tgz"
+
+        def caveats
+          "setuid"
+        end
+      end
+    EOS
+
+    fa.audit_caveats
+    assert_equal ["Don't recommend setuid in the caveats, suggest sudo instead."],
+      fa.problems
+  end
+
+  def test_audit_desc
+    formula_descriptions = [
+      { :name => "foo", :desc => nil,
+        :problem => "Formula should have a desc" },
+      { :name => "bar", :desc => "bar" * 25,
+        :problem => "Description is too long" },
+      { :name => "baz", :desc => "Baz commandline tool",
+        :problem => "Description should use \"command-line\"" },
+      { :name => "qux", :desc => "A tool called Qux",
+        :problem => "Description shouldn't start with an indefinite article" },
+    ]
+
+    formula_descriptions.each do |formula|
+      content = <<-EOS.undent
+        class #{Formulary.class_s(formula[:name])} < Formula
+          url "http://example.com/#{formula[:name]}-1.0.tgz"
+          desc "#{formula[:desc]}"
+        end
+      EOS
+
+      fa = formula_auditor formula[:name], content, :strict => true
+      fa.audit_desc
+      assert fa.problems.empty? # this shouldn't be true, but it is!
```

Comment:
Wow, thanks! I should have double-checked that arithmetic. 😳 

It looks like `"bar: #{'bar' * 25}".length` actually works out to 80, but as you pointed out, it would need to be more than 80 to trigger the warning.

> Additionally, the line below is nicer (and works) as `assert_match formula[:problem], fa.problems.first.to_s`

👍  I've changed it to `assert_match formula[:problem], fa.problems.first`.
(Forgot that `assert_match` would work here!)